sdk/lib/opentelemetry/sdk/internal.rb

```diff
@@ -57,6 +57,10 @@ def valid_attributes?(attrs)
           end
         end
       end
+
+      def valid_exporter?(exporter)
+        exporter && %i[export shutdown force_flush].all? { |m| exporter.respond_to?(m) }
+      end
```

Comment:
It might be useful to move this to the `opentelemetry-common` gem. We provide a couple of implementations of span processors in the SDK, but clients can implement their own, and they could benefit from calling this too.